packages/api/src/sendPayment/sendPayment.ts

```diff
@@ -1,26 +1,28 @@
 import {
   GEM_WALLET,
-  PaymentResponse,
   PaymentRequestPayload,
-  RequestPaymentMessage
+  RequestSendPaymentMessage,
+  SendPaymentResponse
 } from '@gemwallet/constants';
 
 import { sendMessageToContentScript } from '../helpers/extensionMessaging';
 
-export const sendPayment = async (payment: PaymentRequestPayload) => {
+export const sendPayment = async (paymentPayload: PaymentRequestPayload) => {
   /* string: hash of the transaction
    * null: user refused the payment
    * undefined: something went wrong
    */
-  let response: string | null | undefined = undefined;
+  let response: SendPaymentResponse | null | undefined = undefined;
   try {
-    const message: RequestPaymentMessage = {
+    const message: RequestSendPaymentMessage = {
       app: GEM_WALLET,
       type: 'REQUEST_SEND_PAYMENT/V3',
-      payload: payment
+      payload: paymentPayload
+    };
+    const response = await sendMessageToContentScript(message);
+    return {
```

Comment:
How about simply doing: ```return result```
Though I was wondering, in `response.result` don't you have an `id` key transitioning as well?

Reply:
i have put back the previous logic